testing/setup/simulated_backend_wrapper.go

```diff
@@ -2,83 +2,117 @@ package setup
 
 import (
 	"context"
+	"errors"
+	"github.com/ethereum/go-ethereum/rpc"
 	"math/big"
+	"sync"
+
+	protocol "github.com/offchainlabs/bold/chain-abstraction"
 
 	"github.com/ethereum/go-ethereum"
-	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethclient/simulated"
 )
 
 var (
-	_ bind.ContractBackend = &SimulatedBackendWrapper{}
+	_ protocol.ChainBackend = &SimulatedBackendWrapper{}
 )
 
 type SimulatedBackendWrapper struct {
+	lock sync.Mutex
 	*simulated.Backend
 }
 
+func (s *SimulatedBackendWrapper) HeaderNumberUint64(ctx context.Context, number *big.Int) (uint64, error) {
```

Comment:
Interesting. I think that this is always going to get the value "nil" because of the commented out code I mentioned earlier. I wonder if we should just have the implementation of this struct in this testing package always get the latest header.